### PR TITLE
Migrate _log_details.dart to null safety.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/logging/_log_details.dart [hunk 1]
```diff
@@ -25,8 +23,8 @@ class LogDetails extends StatefulWidget {
 
 class _LogDetailsState extends State<LogDetails>
     with SingleTickerProviderStateMixin {
-  String _lastDetails;
-  ScrollController scrollController;
+  String? _lastDetails;
+  ScrollController? scrollController;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/logging/_log_details.dart:27]: late final

File: packages/devtools_app/lib/src/screens/logging/_log_details.dart [hunk 2]
```diff
@@ -45,12 +43,12 @@ class _LogDetailsState extends State<LogDetails>
...
   Future<void> _computeLogDetails() async {
     if (widget.log?.needsComputing ?? false) {
-      await widget.log.compute();
+      await widget.log!.compute();
       setState(() {});
     }
   }
 
-  bool showSimple(LogData log) => log != null && !log.needsComputing;
+  bool showSimple(LogData log) => !log.needsComputing;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/logging/_log_details.dart:51]: looks like this method can be deleted. It is not used anywhere

File: packages/devtools_app/lib/src/screens/logging/_log_details.dart [hunk 3]
```diff
@@ -89,7 +87,11 @@ class _LogDetailsState extends State<LogDetails>
           needsTopBorder: false,
           rightActions: [
             CopyToClipboardControl(
-              dataProvider: disabled ? null : () => log?.prettyPrinted,
+              dataProvider: disabled
+                  ? null
+                  // The parenthesis are actually needed.
+                  // ignore: unnecessary_parenthesis
+                  : (() => log?.prettyPrinted!) as String Function()?,
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/logging/_log_details.dart:94]: you should be able to get rid of the cast if you write:
`dataProvider: disabled ? null : log?.prettyPrinted`

and then change prettyPrinted from a getter to a function.

File: packages/devtools_app/lib/src/screens/logging/_log_details.dart (hunk 1 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/logging/_log_details.dart]: done

File: packages/devtools_app/lib/src/screens/logging/_log_details.dart (hunk 2 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/logging/_log_details.dart:51]: done

File: packages/devtools_app/lib/src/screens/logging/_log_details.dart (hunk 3 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/logging/_log_details.dart:94]: prettyPrinted can return null, but we want function that  returns non-nullable string for dataProvider. 
Do we want to change dataProvider type to return nullable string?

Still converted prettyPrinted to function as it is too complicated to be getter.

Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/screens/logging/_log_details.dart]: Yeah let's change ClipboardDataProvider to return a nullable string. It looks like down the line, ClipboardData.text is nullable anyway, so this should be fine. We might want to ensure we give a default string value for `data.text` in `Clipboard.setData`

Comment (polina-c) [reply packages/devtools_app/lib/src/screens/logging/_log_details.dart]: done

File: packages/devtools_app/lib/src/screens/logging/_log_details.dart [hunk 4]
```diff
@@ -85,11 +85,7 @@ class _LogDetailsState extends State<LogDetails>
           needsTopBorder: false,
           rightActions: [
             CopyToClipboardControl(
-              dataProvider: disabled
-                  ? null
-                  // The parenthesis are actually needed.
-                  // ignore: unnecessary_parenthesis
-                  : (() => log?.prettyPrinted()!) as String Function()?,
+              dataProvider: disabled ? null : () => log?.prettyPrinted(),
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/logging/_log_details.dart:88]: can you write `log?.prettyPrinted` instead of `() => log?.prettyPrinted()`

Comment (polina-c) [reply packages/devtools_app/lib/src/screens/logging/_log_details.dart:88]: done

**Pull request merged**